### DISCUSSION TITLE
Can we send param with zod.parse()

### DISCUSSION THREAD

**Discussion by caglar-yildiz** [colinhacks/zod#1074] [Q&A] [answered]
Hello,
In the below post, I have asked a question but could not get an answer in Stackoverflow. I want to send a variable inside the Zod schema definition so that I can use them to generate more dynamic responses. 
Can anyone pls help me :) 

https://stackoverflow.com/questions/71720669/how-to-generate-error-messages-with-zod-in-different-languages

**Comment by scotttrinh** [comment, marked as answer]
In that case you'll need to make the schema dynamically within the request. A simple pseudo-code example:

```ts
function routeHandler(req) {
  const schema = z.object({
    params: z.object({
      _id: z.string({ required_error: getMessage(req.body.lang) }),
    }),
  });
  // can use the schema here for whatever validation you are trying to do
}
```
